### DISCUSSION TITLE
Python SVM model in C++

### DISCUSSION THREAD

**Discussion by joshua-777** [microsoft/onnxruntime#15748] [Other Q&A]
Hi, I have trained an SVM model in python with 8 features and a label for binary classification. I have also generated an onnx model of the same. 

       import pandas as pd
       from sklearn.model_selection import train_test_split
       from sklearn.svm import SVC
       from sklearn.metrics import accuracy_score, classification_report
       from sklearn.preprocessing import StandardScaler
       from joblib import dump
       from skl2onnx import convert_sklearn
       from skl2onnx.common.data_types import FloatTensorType

       df = pd.read_csv('nothing.csv')

       X = df.iloc[:, :-1] 
       y = df.iloc[:, -1]  

       model = SVC(kernel='linear')
       model.fit(X, y)

       initial_type = [('float_input', FloatTensorType([None, 8]))]
       onnx_model = convert_sklearn(model, initial_types=initial_type)
       with open("svm.onnx", "wb") as f:
           f.write(onnx_model.SerializeToString())

-------
I am trying to run it in onnx runtime in C++ to get the desired classification of real-time features. I used the example of youtube video by Italian cpp that runs a linear regression onnx model in c++. 

    int main(int argc, char* argv[]) {

    Ort::Env env;

    Ort::Session session{ env, LR"(models\linear.onnx)", Ort::SessionOptions{}};

    Ort::AllocatorWithDefaultOptions allocator;
    auto* input_name = session.GetInputName(0, allocator);
    auto* output_name = session.GetOutputName(0, allocator);
    auto input_shape = session.GetInputTypeInfo(0).GetTensorTypeAndShapeInfo().GetShape();
    std::vector<float> input_data{ 9,5,6 };
    auto memoryInfo = Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
    auto inputOnnxTensor = Ort::Value::CreateTensor<float>(memoryInfo, input_data.data(), input_data.size(), input_shape.data(), 
    input_shape.size());
  
    std::vector<const char*> inputNames = { input_name };
    std::vector<const char*> outputNames = { output_name };
    // Run inference
    auto outputValues = session.Run(Ort::RunOptions{ nullptr }, inputNames.data(), &inputOnnxTensor, 1, outputNames.data(), 1);

    // Get the output tensor data
    auto& output1 = outputValues[0];
    const auto* floats = output1.GetTensorMutableData<float>();
    const auto floatsCount = output1.GetTensorTypeAndShapeInfo().GetElementCount();
   
    std::copy_n(floats, floatsCount, std::ostream_iterator<float>(std::cout, " "));
    allocator.Free(input_name);
    allocator.Free(output_name);
    }

It works fine for that model. But when i change the model to the svm onnx model, i change input_data with 8 values for 8 features in order to generate the classification, the code doesn't even run entirely. I am a beginner using onnx runtime, What changes must be made in the code for running the svm model in onnx runtime that accepts 8 features for binary classification. Please help, it is urgent.
